src/SearchWrapper.cpp

```diff
@@ -0,0 +1,163 @@
+#include <Rcpp.h>
+#include <xapian.h>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+using namespace std;
+
+Rcpp::List convertToDataFrame(Rcpp::List list) {
+  Rcpp::List dataFrame = clone(list);
+  Rcpp::GenericVector aRow = dataFrame(0);
+  Rcpp::StringVector rowNames(aRow.length());
+  
+  for (int i = 0; i < aRow.length(); ++i) {
+    char name[5];
+    sprintf(&(name[0]), "%d", i+1);
+    rowNames(i) = name;
+  }
+  
+  dataFrame.attr("row.names") = rowNames;
+  Rcpp::StringVector colNames(dataFrame.length());
+  
+  int ascii=97;
+  for (int j = 0; j < dataFrame.length(); ++j) {
+    char name[6];
+    //sprintf(&(name[0]), "%d", j+1);
+    sprintf(&(name[0]), "%c", char(ascii++));
```

Comment:
I'd consider putting the `ascii` increment into the `for` loop, since it happens every time. You also don't need to convert the integer to a char to be able to use it with the `printf()` family. Something like:

```c++
for (int j=0, ascii=97; j < dataFrame.length(); ++j, ++ascii) {
  char name[6];
  snprintf(name, 6, "%c", ascii);
  colNames(j) = name;
}
```